misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeAction.java

```diff
@@ -78,7 +78,9 @@ public boolean isEnabled(LanguageServerContext serverContext) {
      */
     private static Optional<CodeAction> getAIDataMapperCommand(Diagnostic diagnostic, CodeActionContext context) {
         try {
-            if (CommonUtil.getRawType(context.positionDetails().matchedExprType()).typeKind() == TypeDescKind.RECORD) {
+            TypeDescKind typeDescriptor = CommonUtil.getRawType(context.positionDetails().matchedExprType()).typeKind();
+
+            if (typeDescriptor == TypeDescKind.RECORD || typeDescriptor == TypeDescKind.COMPILATION_ERROR) {
```

Comment:
Isn't the COMPILATION_ERROR kind too generic? what's an example for matching this case?

Reply:
When `check` is used, typeDescriptor is a COMPILATION_ERROR
I.e. - `Grades grades = check getStudent();`